drivers/ble/sphero/sphero_r2d2_driver.go

```diff
@@ -0,0 +1,396 @@
+package sphero
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"gobot.io/x/gobot/v2"
+	"gobot.io/x/gobot/v2/drivers/ble"
+	"gobot.io/x/gobot/v2/drivers/common/spherocommon"
+)
+
+type Playback uint8
+
+type LegAction uint8
+
+const (
+	r2AntiDosChara  = "00020005574f4f2053706865726f2121" // handshake 1st transmission
+	r2CommandsChara = "00010002574f4f2053706865726f2121" // send uuid
+	r2ResponseChara = r2CommandsChara
+
+	// command safe interval
+	commandInterval = time.Duration(12) * time.Millisecond
+
+	// start of packet
+	sop = 0x8D
+	// end of packet
+	eop        = 0xD8
+	escapeHex  = 0xAB
+	escapeMask = 0x88
+
+	// flags
+	isResponse                = 0b1
+	requestsResponse          = 0b10
+	requestsOnlyErrorResponse = 0b100
+	isActivity                = 0b1000
+	hasTargetId               = 0b10000
+	hasSourceId               = 0b100000
+	unused                    = 0b1000000
+	extendedFlags             = 0b10000000
```

Comment:
```suggestion
	isResponse                = 0b00000001
	requestsResponse          = 0b00000010
	requestsOnlyErrorResponse = 0b00000100
	isActivity                = 0b00001000
	hasTargetId               = 0b00010000
	hasSourceId               = 0b00100000
	unused                    = 0b01000000
	extendedFlags             = 0b10000000
```
is maybe more clear, alternatively use hex values